src/realm/query_engine.hpp

```diff
@@ -1680,8 +1684,17 @@ class StringNode : public StringNodeBase {
         TConditionFunction cond;
 
         for (size_t s = start; s < end; ++s) {
+
             StringData t = get_string(s);
 
+            if constexpr (std::is_same_v<TConditionFunction, NotEqual>) {
+                if (m_interned_string) {
+                    const auto id = m_string_interner->lookup(get_string(s));
```

Comment:
We already got the string in `t`. I think the logic here is wrong - we should not call get_string() if we can just get the StringID and compare with that.